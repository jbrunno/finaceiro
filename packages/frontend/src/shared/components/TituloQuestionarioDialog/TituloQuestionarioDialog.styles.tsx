import { styled, css } from '@frontend/styles';
import { CopyToClipboardButton, DialogContent } from '@frontend/components';

export const Header = styled('div')(
  ({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    gap: ${theme.fn.spacing(8)};
    margin: 0 auto ${theme.fn.spacing(16)};
  `,
);

export const CopiarDetalhes = styled('div')(
  ({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr auto;
    gap: ${theme.fn.spacing(8)};
  `,
);

export const DialogContentStyled = styled(DialogContent)`
  width: ${({ theme }) => theme.fn.size(444)};
`;

export const CopyToClipboardButtonStyled = styled(CopyToClipboardButton)`
  margin-top: ${({ theme }) => theme.fn.spacing(18)};
`;
