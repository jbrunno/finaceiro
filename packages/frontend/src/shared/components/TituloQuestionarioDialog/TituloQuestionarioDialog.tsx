import {
  Typography,
  Dialog,
  DialogProps,
  CardText,
  Box,
} from '@frontend/components';
import {
  Header,
  DialogContentStyled,
  CopiarDetalhes,
  CopyToClipboardButtonStyled,
} from './TituloQuestionarioDialog.styles';
import { TituloQuestionario } from '../TituloCard/TituloCard.gql';

type TituloQuestionarioDialogProps = {
  className?: string;
  tituloQuestionario: TituloQuestionario;
} & Omit<DialogProps, 'children'>;

export function TituloQuestionarioDialog({
  tituloQuestionario,
  ...dialogProps
}: TituloQuestionarioDialogProps) {
  return (
    <Dialog {...dialogProps} maxWidth="xl">
      <DialogContentStyled>
        <Header>
          <Typography variant="h3" color="textPrimary">
            Detalhes do título
          </Typography>
        </Header>
        <Box>
          <CopiarDetalhes>
            <CardText
              label="Site FTX de registro"
              text={tituloQuestionario.siteRegistro}
            />
            <CopyToClipboardButtonStyled
              tooltip="Copiar detalhes"
              text={tituloQuestionario.siteRegistro || ''}
              showNotification
              textToast="Informações copiadas para a área de transferência."
            />
          </CopiarDetalhes>
          <CopiarDetalhes>
            <CardText
              label="Conta anterior a data 11/11/2022"
              text={tituloQuestionario.contaAnteriorFalencia ? 'Sim' : 'Não'}
            />
            <CopyToClipboardButtonStyled
              tooltip="Copiar detalhes"
              text={tituloQuestionario.contaAnteriorFalencia ? 'Sim' : 'Não'}
              showNotification
              textToast="Informações copiadas para a área de transferência."
            />
          </CopiarDetalhes>
          <CopiarDetalhes>
            <CardText
              label="Tipo de conta"
              text={tituloQuestionario.tipoConta}
            />
            <CopyToClipboardButtonStyled
              tooltip="Copiar detalhes"
              text={tituloQuestionario.tipoConta || ''}
              showNotification
              textToast="Informações copiadas para a área de transferência."
            />
          </CopiarDetalhes>
          <CopiarDetalhes>
            <CardText
              label="Tipo de ativos utilizados"
              text={tituloQuestionario.tipoAtivoInvestido}
            />
            <CopyToClipboardButtonStyled
              tooltip="Copiar detalhes"
              text={tituloQuestionario.tipoAtivoInvestido || ''}
              showNotification
              textToast="Informações copiadas para a área de transferência."
            />
          </CopiarDetalhes>
          <CopiarDetalhes>
            <CardText
              label="Valor aproximado dos depósitos"
              text={tituloQuestionario.valorDepositos}
            />
            <CopyToClipboardButtonStyled
              tooltip="Copiar detalhes"
              text={tituloQuestionario.valorDepositos || ''}
              showNotification
              textToast="Informações copiadas para a área de transferência."
            />
          </CopiarDetalhes>
          <CopiarDetalhes>
            <CardText
              label="Informações adicionais"
              text={
                <Typography variant="body2">
                  {tituloQuestionario.observacao}
                </Typography>
              }
            />
            <CopyToClipboardButtonStyled
              tooltip="Copiar detalhes"
              text={tituloQuestionario.observacao || ''}
              showNotification
              textToast="Informações copiadas para a área de transferência."
            />
          </CopiarDetalhes>
        </Box>
      </DialogContentStyled>
    </Dialog>
  );
}
