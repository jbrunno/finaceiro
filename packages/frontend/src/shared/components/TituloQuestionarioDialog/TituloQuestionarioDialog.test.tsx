import { render } from '@test-utils/component-testing';
import { TituloQuestionarioDialog } from './TituloQuestionarioDialog';

test('should render TituloQuestionarioDialog', () => {
  const { container } = render(
    <TituloQuestionarioDialog
      open
      onClose={() => {}}
      tituloQuestionario={{
        contaAnteriorFalencia: false,
        observacao:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla quis ligula et felis interdum fermentum in ut quam.',
        siteRegistro: 'https://ftx.com',
        tipoAtivoInvestido: 'Bitcoin',
        tipoConta: 'Conta de bitcoins',
        valorDepositos: 'R$ 1.000,00',
      }}
    />,
  );
  expect(container).toMatchSnapshot();
});
