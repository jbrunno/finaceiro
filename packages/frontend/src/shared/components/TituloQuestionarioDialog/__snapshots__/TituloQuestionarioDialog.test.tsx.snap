// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TituloQuestionarioDialog 1`] = `
<div
  aria-hidden="true"
/>
`;
